### ISSUE TITLE
Generated .g.cs uses non-namespaced `As<T>()` extension method

### ISSUE THREAD

**Issue by mqudsi** [microsoft/microsoft-ui-xaml#5375]
The code generated for WinUI 3 XAML pages in the `Foo.g.cs` uses the non-namespaced WinRT `.As<T>(...)` extension method, but if the project contains a `.As<T>()` extension method in the default project namespace, that will be used in its place.

For example, we always have a utility extension method with the same name that eliminates the need to wrap each `(foo as Foo)` invocation in parentheses by invoking it as object-attached extension method:

```csharp
namespace MySolution {
    static class ObjectExtensions
    {
        public abstract class IClass<T> where T : class
        {
        }

        public abstract class IStruct<T> where T : struct
        {
        }

        // object.As<class>: return null if not possible
        public static Y As<Y>(this object x, IClass<Y> _ = null)
            where Y : class
        {
            return x as Y;
        }

        // object.As<struct>: throw exception if not possible
        public static Y As<Y>(this object x, IStruct<Y> _ = null)
            where Y : struct
        {
            return (Y)(object)x;
        }
    }
}
```

The generated `MainPage.g.cs` under Visual Studio 2019 16.11.0 Preview 2 with Project Reunion 0.8.0 for a Win32 WinUI project looks like this:

```csharp
using WinRT;

namespace MySolution {
...
            public void Connect(int connectionId, global::System.Object target)
            {
                switch(connectionId)
                {
                    case 22: // MainPage.xaml line 58
                        this.obj22 = new global::System.WeakReference(target.As<Microsoft.UI.Xaml.Controls.Grid>());
                        break;
                    case 23: // MainPage.xaml line 86
                        this.obj23 = target.As<Microsoft.UI.Xaml.Controls.TextBlock>();
                        break;
                    case 24: // MainPage.xaml line 94
                        this.obj24 = target.As<Microsoft.UI.Xaml.Controls.TextBlock>();
                        break;
                    case 25: // MainPage.xaml line 102
                        this.obj25 = target.As<Microsoft.UI.Xaml.Controls.TextBlock>();
                        break;
                    case 26: // MainPage.xaml line 82
                        this.obj26 = target.As<Microsoft.UI.Xaml.Media.ImageBrush>();
                        break;
                    case 27: // MainPage.xaml line 72
                        this.obj27 = target.As<Microsoft.UI.Xaml.Media.SolidColorBrush>();
                        break;
                    default:
                        break;
                }
            }
...
}
```

All those `target.As<Foo>()` invocations will use the namespace-local `ObjectExtensions.As<T>()` extension method (in fact, the `using WinRT;` at the top of the generated CS file is greyed out as the reference is altogether unused). For simple object casting, our extension method of the same name suffices, but it does not perform COM interfacing so attempting to use it to cast `target` to a COM32 interface will fail.

One workaround is to use an additional separate namespace for the generated impl and forward all methods in the original generated partial class to a second impl in the separate namespace where they won't automatically pick up the non-imported `As<T>()` extension method, but honestly, given that this is machine generated code, it probably makes more sense to just bite the bullet and explicitly call the WinRT `As<T>()` method a la `WinRT.CastExtensions.As<Foo>(target)` instead of using it as an extension method altogether.

**Comment by StephenLPeters** [comment]
@alwu-msft and @RealTommyKlein FYI

**Comment by RealTommyKlein** [comment]
Thanks for this detailed writeup - agreed we should do an explicit call instead of an extension method.  The typecasts also should use fully qualified namespaces to avoid any ambiguity as well (`global::Microsoft.UI.Xaml.Controls.Grid` instead of `Microsoft.UI.Xaml.Controls.Grid`).

**Comment by RealTommyKlein** [comment]
The fix will be in Reunion 1.0, closing this issue - thanks!
